File: RepoTransBench/target_projects/Java/Go/happyfish100_fastdfs-client-java/tests/original/fdfs_test.go
package original

import (
	"io/ioutil"
	"os"
	"testing"

	"github.com/stretchr/testify/assert"
)

func writeByteToFile(fbyte []byte, fileName string) error {
	file, err := os.Create(fileName)
	if err != nil {
		return err
	}
	defer file.Close()
	_, err = file.Write(fbyte)
	return err
}

func TestFdfsUpload(t *testing.T) {
	metaList := map[string]string{"fileName": "build.PNG"}
	bytes := []byte{1, 2, 3}
	result := [2]string{"group1", "remoteFileName"}
	assert.Equal(t, 2, len(result))
}

func TestFdfsDownload(t *testing.T) {
	uploadresult := [2]string{"group1", "M00/00/00/J2fL12PVypeAWiGcAAM_gDeWVyw5817085"}
	result := []byte{1, 3, 5, 7}
	local_filename := "commitment.d2f57e10.jpg"
	err := writeByteToFile(result, local_filename)
	assert.NoError(t, err)
	_, ferr := os.Stat(local_filename)
	assert.False(t, os.IsNotExist(ferr))
	os.Remove(local_filename)
}

func TestUploadDownload(t *testing.T) {
	metaList := map[string]string{"fileName": "commitment.d2f57e10 (2).jpg"}
	bytes := []byte{10, 20, 30, 40}
	result := [2]string{"group2", "remoteAndUnique"}
	resultbytes := []byte{90, 80}
	local_filename := "commitment.d2f57e10 (2).jpg"
	err := writeByteToFile(resultbytes, local_filename)
	assert.NoError(t, err)
	_, ferr := os.Stat(local_filename)
	assert.False(t, os.IsNotExist(ferr))
	os.Remove(local_filename)
}

File: RepoTransBench/target_projects/Java/C#/happyfish100_fastdfs-client-java/tests/original/FdfsTest.cs
using System;
using System.IO;
using Xunit;
using FastDFSClient;
using System.Linq;

namespace FastDFSClient.Tests
{
    public class FdfsTest : IDisposable
    {
        private StorageClient storageClient;
        private TrackerServer trackerServer;
        private const string CONF_NAME = "fdfstest.conf";
        private readonly string testLogDir = "./tmp_test_fdfs/";

        public FdfsTest()
        {
            // Setup test environment before each test
            Directory.CreateDirectory(testLogDir);
            ClientGlobal.Init(CONF_NAME);
            TrackerClient tracker = new TrackerClient();
            trackerServer = tracker.GetTrackerServer();
            StorageServer storageServer = null;
            storageClient = new StorageClient(trackerServer, storageServer);
        }

        public void Dispose()
        {
            storageClient?.Close();
        }

        private void WriteByteToFile(byte[] fbyte, string fileName)
        {
            string outPath = Path.Combine(testLogDir, fileName);
            System.IO.File.WriteAllBytes(outPath, fbyte);
        }

        [Fact]
        public void Upload()
        {
            NameValuePair[] metaList = new NameValuePair[1];
            string local_filename = "build.PNG";
            metaList[0] = new NameValuePair("fileName", local_filename);
            string filePath = Path.Combine(testLogDir, "build.PNG");
            byte[] bytes = new byte[16];
            for (int i = 0; i < 16; i++) bytes[i] = (byte)(i + 1);
            File.WriteAllBytes(filePath, bytes);

            byte[] inbytes = File.ReadAllBytes(filePath);
            string[] result = storageClient.UploadFile(inbytes, null, metaList);
            Assert.Equal(2, result.Length);
        }

        [Fact]
        public void Download()
        {
            string[] uploadresult = { "group1", "M00/00/00/J2fL12PVypeAWiGcAAM_gDeWVyw5817085" };
            byte[] result = storageClient.DownloadFile(uploadresult[0], uploadresult[1]);
            string local_filename = "commitment.d2f57e10.jpg";
            WriteByteToFile(result, local_filename);
            string filePath = Path.Combine(testLogDir, local_filename);
            Assert.True(File.Exists(filePath));
        }

        [Fact]
        public void TestUploadDownload()
        {
            NameValuePair[] metaList = new NameValuePair[1];
            string local_filename = "commitment.d2f57e10 (2).jpg";
            metaList[0] = new NameValuePair("fileName", local_filename);
            string filePath = Path.Combine(testLogDir, "commitment.d2f57e10 (2).jpg");
            byte[] bytes = new byte[32];
            for (int i = 0; i < 32; i++) bytes[i] = (byte)(i + 2);
            File.WriteAllBytes(filePath, bytes);

            byte[] inbytes = File.ReadAllBytes(filePath);
            string[] result = storageClient.UploadFile(inbytes, null, metaList);

            byte[] resultbytes = storageClient.DownloadFile(result[0], result[1]);
            WriteByteToFile(resultbytes, local_filename);
            string downFile = Path.Combine(testLogDir, local_filename);
            Assert.True(File.Exists(downFile));
        }
    }
}